code-for-canada/project-thundercat
# [Main Nav] Hide main navigation when in test

# Description

Addition of TestStatusRedux, as well as logic to hide main nav and resize all relevant in-test components.
This gives the candidate and the developers more room within the test, and also prevents a user from accidentally exiting the test by clicking on one of the main nav buttons.

NOTE:
- Even though we are removing 3 of the buttons from our main nav, we do not have an empty main nav: the toggle language button is technically part of it, so it is a main nav with one item.
- most of the 97 "Changes" in App.js are do to a change in spacing
- I created a Constants.js file to store header and footer dimensions. I can easily move this to another file, but they seem like baselines that could be used for any test, not just eMIB. but I am open to suggestions for a better place to put them.

## Type of change

- New feature (non-breaking change which adds functionality)
- Code cleanliness or refactor

## Screenshot

No change to other tabs or pre-test pages:
![image](https://user-images.githubusercontent.com/2746350/55423444-e5ee0400-554b-11e9-839f-e02e676a796c.png)

Changes within a test:
![image](https://user-images.githubusercontent.com/2746350/55423483-fd2cf180-554b-11e9-8997-dceb49ef99e8.png)

![image](https://user-images.githubusercontent.com/2746350/55423575-2b123600-554c-11e9-80d2-ee45c42d177e.png)

![image](https://user-images.githubusercontent.com/2746350/55423502-074ef000-554c-11e9-9f33-56f080116fa1.png)

![image](https://user-images.githubusercontent.com/2746350/55423536-1897fc80-554c-11e9-8059-ccccd5cca0e3.png)


# Testing

Applicable for all code changes.

Manual steps to reproduce this functionality:

1.  Go to ProtoType -> Start eMIB
2.  Click through the pre-test instructions -> Start Test
3.  See that the main nav has disappeared completely

# Checklist

Applicable for all code changes.
